src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -64,6 +65,53 @@ std::vector<std::string> get_supported_metrics() {
           "confusion_matrix", "roc_curve"};
 }
 
+size_t count_correct_predictions(size_t num_classes, const shared_float_array& output_chunk, 
+    const shared_float_array& label_chunk, size_t num_predictions) {
+      
+  const float* output_ptr = output_chunk.data();
+  const float* truth_ptr = label_chunk.data();
+    
+
+  size_t num_correct_predictions = 0.f;
```

Comment:
Should initialize a `size_t` variable with an integer literal `0`, not a floating-point literal `0.f`